mlflow/server/handlers.py

```diff
@@ -4572,6 +4573,94 @@ def _get_dataset_records_handler(dataset_id):
 
 
+# Cache for telemetry config with 1 hour TTL
+_telemetry_config_cache = TTLCache(maxsize=1, ttl=3600)
+
+
+@catch_mlflow_exception
+def get_ui_telemetry_handler():
+    """
+    GET handler for /telemetry endpoint.
+    Returns the telemetry client configuration by fetching it directly.
+    """
+    from mlflow.telemetry.utils import (
+        FALLBACK_UI_CONFIG,
+        fetch_ui_telemetry_config,
+        is_telemetry_disabled,
+    )
+
+    if is_telemetry_disabled():
+        return jsonify(FALLBACK_UI_CONFIG)
+
+    if "config" in _telemetry_config_cache:
+        config = _telemetry_config_cache["config"]
+    else:
+        config = fetch_ui_telemetry_config()
+        _telemetry_config_cache["config"] = config
+
+    # UI telemetry should be also disabled if overall telemetry is disabled
+    disable_ui_telemetry = config.get("disable_ui_telemetry", True) or config.get(
+        "disable_telemetry", True
+    )
+    response = {
+        "disable_ui_telemetry": disable_ui_telemetry,
+        "disable_ui_events": config.get("disable_ui_events", []),
+        "ui_rollout_percentage": config.get("ui_rollout_percentage", 0),
+    }
+    return jsonify(response)
+
+
+@catch_mlflow_exception
+def post_ui_telemetry_handler():
+    """
+    POST handler for /telemetry endpoint.
+    Accepts telemetry records and adds them to the telemetry client.
+    """
+    from mlflow.telemetry import get_telemetry_client
+    from mlflow.telemetry.schemas import Record, Status
+    from mlflow.telemetry.utils import fetch_ui_telemetry_config, is_telemetry_disabled
+
+    if is_telemetry_disabled():
+        return jsonify({"status": "disabled"})
+
+    data = request.json.get("records", [])
+
+    # check cached config to see if telemetry is disabled
+    # if so, don't process the records. we don't rely on the
+    # config from the telemetry client because it is only fetched
+    # once, so it won't be updated unless the server is restarted.
+    if "config" in _telemetry_config_cache:
+        config = _telemetry_config_cache["config"]
+    else:
+        config = fetch_ui_telemetry_config()
+        _telemetry_config_cache["config"] = config
+
+    # if updated telemetry config is disabled / missing, tell the UI to stop sending records
+    if config.get("disable_ui_telemetry", True) or config.get("disable_telemetry", True):
+        return jsonify({"status": "disabled"})
+
+    records = [
+        Record(
+            event_name=event["event_name"],
+            timestamp_ns=event["timestamp_ns"],
+            params=event["params"],
+            status=Status.SUCCESS,
+            installation_id=event["installation_id"],
+            session_id=event["session_id"],
+            duration_ms=0,
+        )
+        for event in data
+    ]
+
+    # Add record to telemetry client
+    client = get_telemetry_client()
+    if client is not None:
+        for record in records:
+            client.add_record(record)
```

Comment:
Could you add `add_records` method in telemetry client for this? we use add_record in sdk because each record is added separately, but pending the whole records all at once is easier in your case